src/frontend/js/components/DashBoard/routes.tsx

```diff
@@ -0,0 +1,44 @@
+import { RouteObject } from 'react-router-dom';
+import { defineMessages, MessageDescriptor } from 'react-intl';
+
+export interface DashBoardRoute extends RouteObject {
+  intlPath?: MessageDescriptor;
+  show: () => boolean;
+  intlTitle: MessageDescriptor;
+  title?: string;
+  protected?: boolean;
+}
+
+export const messages = defineMessages({
+  coursesDashBoardRouteTitle: {
+    id: 'components.Dashboard.routes.courses.title',
+    description: '',
+    defaultMessage: 'Courses',
+  },
+  coursesDashBoardRoutePath: {
+    id: 'components.Dashboard.routes.courses.path',
+    description: '',
+    defaultMessage: 'course',
+  },
+  preferencesDashBoardRouteTitle: {
+    id: 'components.Dashboard.routes.preferences.title',
+    description: '',
+    defaultMessage: 'My preferencess',
+  },
+});
+
+const dashboardRoutes: Array<DashBoardRoute> = [
+  {
+    element: <h2>Courses List</h2>,
+    intlPath: messages.coursesDashBoardRoutePath,
+    intlTitle: messages.coursesDashBoardRouteTitle,
+    show: () => true,
```

Comment:
You did not use it anywhere. I think, at first we do not need this property so remove it then if a feature is required about that, this is will be an easy pick.
```suggestion
    show: () => true,
```

Reply:
I removed it, then reimplemented it, cause i do think we neeed it for the 404 page purpose,...
But, by default this time the link is showed, and, if the property is defined, then it is shown depending of the value returned by the show property